src/pages/index.js

```diff
@@ -158,50 +177,120 @@ closeButtons.forEach((button) => {
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitButton, false);
+    });
 }
 
 editFormElement.addEventListener("submit", handleEditFormSubmit);
 
+editAvatarForm.addEventListener("submit", handleAvatarSubmit);
+
 newPostButton.addEventListener("click", () => {
   openModal(newPostModal);
 });
 
-// postCloseButton.addEventListener("click", () => {
-//   closeModal(newPostModal);
-// });
+postFormElement.addEventListener("submit", handleAddCard);
 
-// imageViewCloseButton.addEventListener("click", () => {
-//   closeModal(imageViewModal);
-// });
+deleteCardForm.addEventListener("submit", handleDeleteSubmit);
 
-postFormElement.addEventListener("submit", handleNewPostSubmit);
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+
+  const submitButton = evt.submitter;
+  setButtonText(submitButton, true);
+
+  api
+    .editAvatarInfo({ avatar: editAvatarInput.value })
+    .then((data) => {
+      avatarImage.src = data.avatar;
+      closeModal(editAvatarModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitButton, false);
+    });
+}
 
-function handleNewPostSubmit(evt) {
+function handleAddCard(evt) {
   evt.preventDefault();
-  console.log(postLinkInput.value);
-  console.log(postCaptionInput.value);
-  const inputVariables = {
-    link: postLinkInput.value,
-    name: postCaptionInput.value,
-  };
-
-  const cardElement = getCardElement(inputVariables);
-  cardsList.prepend(cardElement);
-  closeModal(newPostModal);
-  evt.target.reset();
-  disableButton(cardSubmitButton, settings);
+
+  const submitButton = evt.submitter;
+  setButtonText(submitButton, true);
+
+  api
+    .addCard({ name: postCaptionInput.value, link: postLinkInput.value })
+    .then((cardData) => {
+      const cardElement = getCardElement(cardData);
+      cardsList.prepend(cardElement);
+      closeModal(newPostModal);
+      postFormElement.reset();
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitButton, false);
+    });
 }
 
-initialCards.forEach(function (i) {
-  const cardElement = getCardElement(i);
-  cardsList.append(cardElement);
-});
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+
+  const loadingText = "Deleting...";
+  const defaultText = "Delete";
+  const submitButton = evt.submitter;
+  setButtonText(submitButton, true, loadingText, defaultText);
+
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteCardModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitButton, false, loadingText, defaultText);
+    });
+}
+
+function handleCardDelete(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteCardModal);
+}
 
 const modal = document.querySelectorAll(".modal");
```

Comment:
If you find DOM elements with `querySelectorAll`, you get a list of `DOM` elements. So you need to name the variable in **plural** (there is such a point in the checklist)